src/snarky.d.ts

```diff
@@ -43,11 +45,19 @@ declare interface ProvablePure<T> extends Provable<T> {
   check: (x: T) => void;
 }
 
+// ocaml types
+type MlTuple<X, Y> = [0, X, Y];
```

Comment:
is there a reason why OCaml arrays always start with a leading 0?

Reply:
I would guess it is to mimic the memory layout of the native compiler. The native Caml runtime will represent tuples, records and arrays with block with a tag `0`, documentation [here](https://dev.realworldocaml.org/runtime-memory-layout.html).
> Tuples, records, and arrays are all represented identically at runtime as a block with tag 0. Tuples and records have constant sizes determined at compile time, whereas arrays can be of variable length. While arrays are restricted to containing a single type of element in the OCaml type system, this is not required by the memory representation.     

And in the JS runtime, it seems it is mimic (tuple, records and arrays are identical, i.e. a JS array with the first element set to `0`.